### PR TITLE
Initial work on the donation cancellation usecase

### PR DESCRIPTION
**Pull request by JeroenDeDauw**

### CODE REVIEW

File: app/routes.php [hunk 1]
```diff
@@ -164,4 +165,20 @@ function( Request $request ) use ( $app, $ffFactory ) {
 	}
 );
 
+$app->post(
+	'donation/cancel',
+	function( Request $request ) use ( $app, $ffFactory ) {
+		$cancellationRequest = new CancelDonationRequest(
+			$request->get( 'firstname', 'sid' ),
+			$request->get( 'firstname', 'token' ),
```
Comment (gbirke) [inline app/routes.php]: you probably need to change 'firstname' ;-)

Comment (JeroenDeDauw) [inline app/routes.php]: Typical. 3 lines of code that are not tested, so of course they are broken :)

**Comment by JeroenDeDauw** [comment]
I realize this does not do anything yet, though would still be nice if it could be merged already

**Comment by JeroenDeDauw** [comment]
Rebased

**Comment by JeroenDeDauw** [comment]
Rebased again :)

**Comment by JeroenDeDauw** [comment]
And again >_>

File: tests/System/Routes/CancelDonationRouteTest.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+				'sid' => '',
+				'token' => '',
+				'utoken' => '',
+			]
+		);
+
+		$this->assertSame( 200, $client->getResponse()->getStatusCode() );
+	}
+
+	public function testGivenGetRequest_resultIs405() {
```
Comment (gbirke) [inline tests/System/Routes/CancelDonationRouteTest.php]: Maybe name it `testGivenGetRequest_resultIsMethodNotAllowed`. i had to think of which status code this is.

File: src/UseCases/CancelDonation/CancelDonationRequest.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	public $donationUpdateToken;
+
+	public function __construct( string $donationId, string $donationToken, string $donationUpdateToken ) {
+		$this->donationId = $donationId;
+		$this->donationToken = $donationToken;
+		$this->donationUpdateToken = $donationUpdateToken;
+	}
+
+	public function getLimit(): string {
+		return $this->limit;
```
Comment (gbirke) [inline src/UseCases/CancelDonation/CancelDonationRequest.php]: This should probably be changed to `donationId` related stuff?

Comment (JeroenDeDauw) [inline src/UseCases/CancelDonation/CancelDonationRequest.php]: Yeah... definitely should have had some more discipline when creating this commit. This is the second defect.

**Pull request merged**